landoapi/stacks.py

```diff
@@ -42,35 +38,21 @@ def build_stack_graph(
         between two nodes. `child` and `parent` are also string
         PHIDs.
     """
-    phids = set()
-    new_phids = {revision_phid}
+    result = phab.call_conduit(
+        "differential.revision.search",
+        constraints={"phids": [revision_phid]},
+        limit=1,
+    )
+
+    stack_graph = result["data"][0]["fields"]["stackGraph"]
+    phids = set(stack_graph.keys())
     edges = []
 
-    # Repeatedly request all related edges, adding connected revisions
-    # each time until no new revisions are found.
-    while new_phids:
-        phids.update(new_phids)
-        edges = phab.call_conduit(
-            "edge.search",
-            types=["revision.parent", "revision.child"],
-            sourcePHIDs=[phid for phid in phids],
-            limit=10000,
-        )
-        edges = phab.expect(edges, "data")
-        new_phids = set()
-        for edge in edges:
-            new_phids.add(edge["sourcePHID"])
-            new_phids.add(edge["destinationPHID"])
-
-        new_phids = new_phids - phids
-
-    # Treat the stack like a commit DAG, we only care about edges going
-    # from child to parent. This is enough to represent the graph.
-    edges = {
-        (edge["sourcePHID"], edge["destinationPHID"])
-        for edge in edges
-        if edge["edgeType"] == "revision.parent"
-    }
+    for source_node, destination_nodes in stack_graph.items():
+        for predecessor_node in destination_nodes:
+            edges.append((source_node, predecessor_node))
+
+    edges = set(edges)
```

Comment:
We can just make `edges` a set to begin with and `.add` the nodes instead of converting from a list.

Reply:
Good idea.